### ISSUE TITLE
Breaking changes and API changes for 1.19

### ISSUE THREAD

**Issue by KnightMiner** [SlimeKnights/TinkersConstruct#5064]
To anyone discovering this issue looking for news about 1.19, see [the FAQ](https://github.com/SlimeKnights/TinkersConstruct/wiki/General-FAQ#what-versions-of-minecraft-do-you-support). This issue is simply a TODO list, not an indicator of update progress.

----

## API changes that can be implemented on 1.18.2

These changes need to be done before a 1.19 release, but can be implemented on 1.18 with deprecations instead of removals

* [x] Migrate all non-essential methods on `Modifier` to `ModifierHook`.
* [x] Implement composable dynamic modifier loader
* [x] Fix or remove EFLN
* [x] Have toolstack stop deleting item stack hide flags
* [x] Make slime islands proper features, with a base feature and configured features, instead of 5 hardcoded base features
* [ ] Look into using GameTest for automated tests, not sure how long our method will keep working as mojang changes bootstrap (plus world sensitive tests would be nice to have)

## Breaking changes for 1.19

* [ ] Books: Remove all uses of named section transformers in favor of generic section data transformers
* [x] Remove `ValidatedResult` in favor of `RecipeResult`, Consider keeping the original name, the generics is the big thing
* [x] Ditch fixed materials in `ToolBuildHandler`, we have a better method now and its a lot of extra code
* [x] Cleanup unneeded methods on various `ResourceLocation` subclasses now that `IdParser` is a thing.

### Modifiers
* [x] Fully migrate tool definition data to hooks
* [ ] Support merging on tool definition data
* [x] Move modifier restrictions to part of the modifier definition
* [x] Remove the old modifier hook system entirely
* [x] Remove all hooks that have been migrated to `ModifierHook`s
* [x] Remove redundant dynamic modifiers
* [x] Make `validate` return `Component` instead of `ValidatedResult`
* [x] Migrate durability bars to use the 0-13 vanilla ranges
* [ ] Narrow return types on `ToolStack` methods to use `NamespaceNBT` instead of `ModDataNBT` when reasonable
* [x] Add starting tool modifier slots to volatile data instead of persistent data, far more stable
* [x] Remove toolbelt NBT migration
* [x] Rename `UnarmedModifier` to `AmbedextrousModifier`
* [x] Remove old options in the modifier serializer
* [x] Remove old aqua affinity hooks

### Tags

* [x] Rename `tconstruct:modifiable/unarmed` to `tconstruct:modifiable/melee/unarmed`
* [x] Rename `tconstruct:modifiable/melee` to `tconsturct:modifiable/melee/held`
* [x] Rename `tconstruct:modifiable/melee_or_unarmed` to `tconstruct:modifiable/melee`
* [x] Rename `tconstruct:modifiable/staffs` to `tconstruct:modifiable/ranged/staffs`

### Resources

* [x] Cleanup the number of factory interfaces on recipes, notably casting recipes
* [x] Replace predicates with the Mantle classes, generics prevent doing so now
* [x] Consider recipe class renames for clarity, since we have a lot more types now
* [x] Remove condition backwards compat from `ModifierSetWorktableRecipe`
* [x] Decide whether I like the tinker station or modifier workstation handling of result tools better, and be consistent

## New forge features on 1.19 to use

* [x] Enchantment override hooks on item stacks as a modifier hook
* [x] Fluid API (fluids will finally do something!)

**Comment by KnightMiner** [comment]
Closing as 1.19 is for the most part done. Moved anything unfinished to #5281
